### PR TITLE
feat: Initial Notifications Model

### PR DESCRIPTION
**Pull request by vinnyjth**
There's not much here yet, just an initial basis to build further off of. No way to test what I've built besides starting the API and checking for changes in your Database, and looking at the tests. 

image

### CODE REVIEW

File: packages/apollos-data-connector-postgres/src/notifications/__tests__/dataSource.tests.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import { sequelize, sync } from '../../postgres/index';
+import { createModel, setupModel } from '../model';
+import { createModel as createPeopleModel } from '../../people/model';
+import NotificationsDataSource from '../dataSource';
+
+let person1;
+
+const context = {};
+
+describe('Apollos Postgres Comment Flags DataSource', () => {
```
Comment (redreceipt) [inline packages/apollos-data-connector-postgres/src/notifications/__tests__/dataSource.tests.js:10]: different description here?

**Comment by redreceipt** [comment]
so local PG is working for me to explore the DB with?

File: packages/apollos-data-connector-postgres/src/notifications/__tests__/dataSource.tests.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-data-connector-postgres/src/notifications/__tests__/dataSource.tests.js:10]: lol, thanks

Comment (vinnyjth) [inline packages/apollos-data-connector-postgres/src/notifications/__tests__/dataSource.tests.js:10]:
```suggestion
describe('Apollos Postgres Notifications DataSource', () => {
```

File: packages/apollos-data-connector-postgres/src/notifications/__tests__/__snapshots__/dataSource.tests.js.snap [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Apollos Postgres Comment Flags DataSource throws an error with an invalid delivery method 1`] = `[Error: new-method is not a valid delivery method]`;
+
+exports[`Apollos Postgres Comment Flags DataSource throws an error with an invalid type 1`] = `[Error: Passing a valid notification type is required for sending a notification]`;
+
+exports[`Apollos Postgres Comment Flags DataSource throws an error without personId 1`] = `[Error: personId is required for sending a notification]`;
```
Comment (vinnyjth) [inline packages/apollos-data-connector-postgres/src/notifications/__tests__/__snapshots__/dataSource.tests.js.snap:7]:
```suggestion
exports[`Apollos Postgres Notifications DataSource throws an error without personId 1`] = `[Error: personId is required for sending a notification]`;
```

File: packages/apollos-data-connector-postgres/src/notifications/__tests__/__snapshots__/dataSource.tests.js.snap [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Apollos Postgres Comment Flags DataSource throws an error with an invalid delivery method 1`] = `[Error: new-method is not a valid delivery method]`;
+
+exports[`Apollos Postgres Comment Flags DataSource throws an error with an invalid type 1`] = `[Error: Passing a valid notification type is required for sending a notification]`;
```
Comment (vinnyjth) [inline packages/apollos-data-connector-postgres/src/notifications/__tests__/__snapshots__/dataSource.tests.js.snap:5]:
```suggestion
exports[`Apollos Postgres Notifications DataSource throws an error with an invalid type 1`] = `[Error: Passing a valid notification type is required for sending a notification]`;
```

File: packages/apollos-data-connector-postgres/src/notifications/__tests__/__snapshots__/dataSource.tests.js.snap [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Apollos Postgres Comment Flags DataSource throws an error with an invalid delivery method 1`] = `[Error: new-method is not a valid delivery method]`;
```
Comment (vinnyjth) [inline packages/apollos-data-connector-postgres/src/notifications/__tests__/__snapshots__/dataSource.tests.js.snap:3]:
```suggestion
exports[`Apollos Postgres Notifications DataSource throws an error with an invalid delivery method 1`] = `[Error: new-method is not a valid delivery method]`;
```

**Comment by vinnyjth** [comment]
> so local PG is working for me to explore the DB with?

I'm not sure I understand the question :)

**Comment by redreceipt** [comment]
> > so local PG is working for me to explore the DB with?
> 
> I'm not sure I understand the question :)

you're looking through the DB with that app above, I thought we were forced to use the remote DB for now?

**Comment by redreceipt** [comment]
I probably just need to get more acquainted with our DB honestly

**Comment by vinnyjth** [comment]
Oh gotcha! Yah - so the way it works is that the local DB will be built correctly, but it will never be filled with data from Rock at the moment.

**Comment by redreceipt** [comment]
once I can get the local DB up and running and connect to it, I'd feel much better about approving DB PRs

**Comment by redreceipt** [comment]
can't connect yet, but maybe there's something I need to do with SSL?
Screen Shot 2021-05-18 at 4 59 07 PM

**Comment by redreceipt** [comment]
@vinnyjth now this is what I'm getting trying to test this
Screen Shot 2021-05-21 at 11 59 37 AM

**Comment by vinnyjth** [comment]
er, I'm not sure how this is supposed to work 🙃, try changing your database name to `postgres` from `michael`

**Comment by vinnyjth** [comment]
It's a catch 22, if you use a DATABASE_URL, it doesn't setup the local database for you. If you don't use the DATABASE_URL, postgres is disabled but it does setup a local database for you.

**Comment by redreceipt** [comment]
Seems like it needs to not try and set up a database if you didn’t set the URL, and if you did, just connect to that one

**Comment by vinnyjth** [comment]
@redreceipt But for local databases, it does need to do some setup if you are connecting for the first time :/

**Comment by vinnyjth** [comment]
You can do this for now. Comment out your local url. start the server. Stop the server. Add your local url back in. Start the server.

**Comment by redreceipt** [comment]
That's what I did, this was the output

Screen Shot 2021-05-21 at 12 11 35 PM

**Comment by vinnyjth** [comment]
@redreceipt can you send me your DATABASE_URL you were using?

**Comment by redreceipt** [comment]
postgres://localhost

**Comment by vinnyjth** [comment]
try
`postgres://localhost/postgres`

that's specifying the DB as `postgres`, instead of the default, which is your macos username.

**Comment by redreceipt** [comment]
Ok that worked, I think for the first time ever. So it only looks for a DB called “postgres” and fails if not?

**Comment by redreceipt** [comment]
I still don't see the notifications table from this PR though

Screen Shot 2021-05-21 at 2 48 25 PM

**Comment by vinnyjth** [comment]
It will only look for whatever DB you specify. If you don't specify, it defaults to the postgres default, which is your username. Normally a DB is already created (like on heroku / aws). Locally, we do the dirty work of building up a default database, but we naively assume you will use database named postgres (I'm assuming because it wasn't clear how to get a sane default for a default db name)

**Comment by vinnyjth** [comment]
Ah shoot, you'll need the companion PR :/

**Comment by vinnyjth** [comment]
https://github.com/ApollosProject/apollos-templates/pull/371

**Comment by redreceipt** [comment]
this is what I get using that PR branch 

Screen Shot 2021-05-21 at 3 07 53 PM

**Comment by vinnyjth** [comment]
Huh. try logging `Notifications`? It seems like the data object is null.

**Comment by redreceipt** [comment]
`undefined`

it's like the symlinks aren't working but when inspecting the node modules, it show's the right thing

Screen Shot 2021-05-21 at 3 17 33 PM

**Comment by vinnyjth** [comment]
Have you tried restarting your server? I've noticed things can be weird when adding/deleting new files.

**Comment by redreceipt** [comment]
yea so strange, `yarn` is definitely having trouble. this is me logging the postgres package

```
ApollosApi % yarn && yarn start:dev                                                                                                                      (notifications) apollos-apps 
yarn install v1.22.10
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
success Already up-to-date.
$ yarn build
yarn run v1.22.10
$ NODE_ENV=production rm -rf ./lib && mkdir -p lib && babel ./src -d ./lib
Successfully compiled 10 files with Babel (641ms).
✨  Done in 0.83s.
✨  Done in 1.53s.
yarn run v1.22.10
$ linkemon ./src/index.js --exec babel-node --delay 2 -e js,yaml,json,yml
[nodemon] 1.18.7
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-cloudinary/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-algolia-search/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-babel-preset-apollos /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-config/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-schema/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-church-online/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-analytics/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-twilio/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-redis-cache/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-server-core/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-eslint-config/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-postgres/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-apollo-server-env-mock/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-rock-apollo-data-source/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-passes/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-onesignal/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-rock/**/* /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-data-connector-bible/**/*
[nodemon] starting `babel-node ./src/index.js`
{
  defineModel: [Getter],
  configureModel: [Getter],
  sequelize: [Getter],
  sync: [Getter],
  PostgresDataSource: [Getter],
  createMigrationRunner: [Getter],
  Comment: {
    dataSource: [Getter],
    schema: [Getter],
    resolver: [Getter],
    migrations: [Getter],
    models: {
      createModel: [Function],
      Visibility: [Object],
      setupModel: [Function]
    }
  },
  UserFlag: {
    dataSource: [Getter],
    migrations: [Getter],
    schema: [Getter],
    models: { createModel: [Function], setupModel: [Function] }
  },
  UserLike: {
    dataSource: [Getter],
    migrations: [Getter],
    models: { createModel: [Function], setupModel: [Function] }
  },
  Follow: {
    dataSource: [Getter],
    schema: [Getter],
    resolver: [Getter],
    migrations: [Getter],
    models: {
      createModel: [Function],
      setupModel: [Function],
      FollowState: [Object]
    }
  },
  Person: {
    dataSource: [Getter],
    schema: [Getter],
    resolver: [Getter],
    migrations: [Getter],
    models: { createModel: [Function] }
  },
  Campus: {
    dataSource: [Getter],
    migrations: [Getter],
    models: { createModel: [Function], setupModel: [Function] }
  }
}
/Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-server-core/lib/index.js:136
}, datum => datum.models);
                  ^

TypeError: Cannot read property 'models' of undefined
    at /Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-server-core/lib/index.js:136:19
    at /Users/michael/Documents/Projects/apollos/apollos-apps/node_modules/lodash/lodash.js:13469:38
    at /Users/michael/Documents/Projects/apollos/apollos-apps/node_modules/lodash/lodash.js:4967:15
    at baseForOwn (/Users/michael/Documents/Projects/apollos/apollos-apps/node_modules/lodash/lodash.js:3032:24)
    at mapValues (/Users/michael/Documents/Projects/apollos/apollos-apps/node_modules/lodash/lodash.js:13468:7)
    at getDbModels (/Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-server-core/lib/index.js:134:51)
    at setupSequelize (/Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-server-core/lib/index.js:172:18)
    at createApolloServerConfig (/Users/michael/Documents/Projects/apollos/apollos-apps/packages/apollos-server-core/lib/index.js:348:3)
    at Object.<anonymous> (/Users/michael/Documents/Projects/apollos/apollos-apps/ApollosApi/src/data/index.postgres.js:96:46)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
[nodemon] app crashed - waiting for file changes before starting...
^C
```

**Comment by redreceipt** [comment]
ooooo i need to build i think

**Comment by redreceipt** [comment]
boom
Screen Shot 2021-05-21 at 3 28 16 PM

**Review by redreceipt — APPROVED** [review]

**Comment by vinnyjth** [comment]
oh good call! I would love to solve that..... somehow

**Comment by redreceipt** [comment]
i did

**Comment by vinnyjth** [comment]
@redreceipt Nice. That will work for the initial boot, but we'll need to do something different for ongoing changes.

**Pull request merged**